dashboard/test/ui/features/project_gallery.feature

```diff
@@ -0,0 +1,45 @@
+# TODO (Erin B.) remove ?showFeatured=1 param from project gallery url during flag removal.
+
+@dashboard_db_access
+Feature: Public Project Gallery
+
+  Background:
+    Given I create a teacher named "Project_Czar"
+    And I give user "Project_Czar" project validator permission
+
+Scenario: Published Projects Show In Recency Order
+  Then I make a playlab project named "Older Published"
+  Then I publish the project
+  Given I am on "http://studio.code.org/projects/public"
+  Then I wait until element ".project_card" is in the DOM
+  Then I wait until element "#ui-project-name" is in the DOM
+  Then I wait until the first "#ui-project-name" contains text "Older Published"
+  Then I make a playlab project named "Newer Published"
+  Then I publish the project
+  Given I am on "http://studio.code.org/projects/public"
+  Then I wait until the first "#ui-project-name" contains text "Newer Published"
+
+Scenario: Featured Projects Show Before Published Projects
+  Then I make a playlab project named "First Featured"
+  Then I publish the project
+  Then I press "#feature_project" using jQuery
+  Given I am on "http://studio.code.org/projects/public?showFeatured=1"
+  Then I wait until element ".project_card" is in the DOM
+  Then I wait until element "#ui-project-name" is in the DOM
+  Then I wait until the first "#ui-project-name" contains text "First Featured"
+  Then I make a playlab project named "Published, NOT Featured"
+  Then I publish the project
+  Given I am on "http://studio.code.org/projects/public?showFeatured=1"
+  Then I wait until element ".project_card" is in the DOM
+  Then I wait until element "#ui-project-name" is in the DOM
+  Then I wait until the first "#ui-project-name" contains text "First Featured"
+
+Scenario: UnPublished, Featured Projects Do Not Show
+  Then I make a playlab project named "Unpublished, Featured"
+  Then I press "#feature_project" using jQuery
+  Given I am on "http://studio.code.org/projects/public?showFeatured=1"
+  Then I wait until element ".project_card" is in the DOM
+  Then I wait until element "#ui-project-name" is in the DOM
+  Then I wait until the first "#ui-project-name" contains text "First Featured"
```

Comment:
Are scenarios always guaranteed to run in order?

Reply:
I believe scenarios are currently run in order, however there's no guarantee that we won't try to change this in the future. For example, in the past I believe @wjordan was considering changing it to only rerun failed scenarios. Ideally, each scenario would not depend on any other scenarios. In this case, it seems easy to fix, by creating another "Published, Featured" project before the "Unpublished, Featured" project and then checking for "Published, Featured" on this line, so that is what I'd recommend doing.